server/master/src/main/java/org/apache/accumulo/master/tableOps/MoveExportedFiles.java

```diff
@@ -48,17 +49,22 @@
 
       Map<String,String> fileNameMappings = PopulateMetadataTable.readMappingFile(fs, tableInfo);
 
+      FileStatus[] exportedFiles = fs.listStatus(new Path(tableInfo.exportDir));
+      FileStatus[] importedFiles = fs.listStatus(new Path(tableInfo.importDir));
+
```

Comment:
Could possibly avoid the linear search in each loop iteration with something like the following before the loop.

```java
Set<String> exporting = Arrays.stream(exportedFiles).map(fstat ->
            fstat.getPath().getName()).map(fileNameMapping::get).collect(Collectors.toSet());
Set<String> exported = Arrays.stream(importedFiles).map(fstat ->
            fstat.getPath().getName()).collect(Collectors.toSet());

//ensure Sets.union(exporting, exported) equals new HashSet<>(fileNameMappings.values)
```

Reply:
This seems quite elegant, and I am working toward this. The only question I have at this stage is the usage of the Guava Sets class. I presume there are no concerns (both for this PR and from later porting this to 2.x) on using Sets here? I did try using CollectionUtils but it seems far easier to perform set operations with the Sets class.

Reply:
Guava Sets is definitely better over commons-collections CollectionUtils. We still use Guava in 2.x, so that should be fine. If you want a pure-Java way to do it, you could use Streams:

```java
Stream.concat(importing.stream(),imported.stream()).collect(Collectors.toSet()).equals(Set.of(fileNameMappings.values()));
```

Guava's `Sets.union` is cleaner.

Reply:
One potential problem with using Guava is the `@Beta` APIs.  Those should be avoided, but its so easy to use them.  I forget to check.  I am not sure if Sets or any of its methods are `@Beta`.

Reply:
Thanks for the tip! I used the [guava-beta-checker](https://github.com/google/guava-beta-checker) plugin to quickly establish if there is any new impact introduced by the usage of Sets. Thankfully, there is none, but in the process, I noticed there are many other @Beta APIs used by previously existing code which are flagged by the plugin. For example, the use of Stopwatch and RateLimiter classes (and others) is flagged. I think that is a separate cleanup project on its own!

Reply:
I can create a separate PR with the guava-beta-checker added and emitting WARN for such beta APIs.

Reply:
I did check. Sets is okay.

Reply:
@arvindshmicrosoft  I found #298.

Reply:
Actually #340 would be the one to look at.

Reply:
Great, so looks like we are fine for the purposes of this PR?

Reply:
guava-beta-checker requires errorprone. I'm not sold on the idea of dictating the compiler in the build, especially not for the default build. The spotbugs plugin would be less invasive (not sure that's the right word I mean). In any case, that can be a separate issue, if it matters.

I'm going to make one more pass with some comments.